### ISSUE TITLE
Improve PerspectiveCamera rotation (upvector)

### ISSUE THREAD

**Issue by louisponet** [JuliaGL/GLAbstraction.jl#76]
https://github.com/JuliaGL/GLAbstraction.jl/blob/7f20d624abac28c333ca22922803b03138c293e6/src/GLCamera.jl#L496

Hi,
I think `Vec3f(0,0,1)` in this line should be changed to `up_v`. Previously the camera remained with the same upvector, causing left/right rotations to feel very weird after a while. 
(also this was what I mentioned in my last comment in  https://github.com/JuliaGL/GLVisualize.jl/issues/190 , I edited that comment) 

grtz

**Comment by jaakkor2** [comment]
Any reason not to do the change?

I agree that if up-vector flips, then left-right rotations do not feel natural.

**Comment by SimonDanisch** [comment]
Yes I think so, I should have added a comment there since I don't completely remember myself... But I think the rotations was getting super funky without a fixed upvector.

**Comment by louisponet** [comment]
If it's of any worth, I changed it locally on my pc since my post and haven't noticed any particular funkyness. Maybe I'm missing something though.
